manifests/plugin/remote_execution/ssh/params.pp

```diff
@@ -0,0 +1,10 @@
+# Remote Execution SSH default parameters
+class foreman_proxy::plugin::remote_execution::ssh::params {
+  $enabled           = true
+  $listen_on         = 'https'
+  $generate_keys     = true
+  $ssh_identity_dir  = "${::foreman_proxy::dir}/.ssh"
```

Comment:
I was afraid of strict variables. In that case I agree (2) is the better choice.